tests/unit/pages/Editor.spec.js

```diff
@@ -1841,4 +1843,60 @@ describe("Editor.vue", () => {
     expect(maximizeModal).toHaveBeenCalled();
     expect(wrapper.vm.isModalMinimized).toBe(false);
   });
+
+  it("does not interfere with irrelevant dialog confirm trigger", async () => {
+    wrapper = mount(Editor, {
+      data() {
+        return {
+          videoId: "abcdefgh",
+        };
+      },
+    });
+
+    await wrapper.find('[data-test="publishButton"]').trigger("click");
+
+    // change the dialog action so that it is no longer
+    // relevant to this component
+    const newDialogAction = "testAction";
+    await store.dispatch("dialog/setDialogAction", newDialogAction);
+
+    // simulate clicking the confirm button of the dialog box
+    await simulateConfirmClick(); // deepscan-disable-line
+
+    // the dialog action shouldn't have been affected and
+    // the confirm click status should remain active
+    expect(wrapper.vm.isDialogConfirmClicked).toBeTruthy();
+    expect(wrapper.vm.dialogAction).toBe(newDialogAction);
+
+    // reset dialog confirm clicked status
+    await store.dispatch("dialog/unsetConfirmClicked");
+  });
+
+  it("does not interfere with irrelevant dialog cancel trigger", async () => {
+    wrapper = mount(Editor, {
+      data() {
+        return {
+          videoId: "abcdefgh",
+        };
+      },
+    });
+
+    await wrapper.find('[data-test="publishButton"]').trigger("click");
+
+    // change the dialog action so that it is no longer
+    // relevant to this component
+    const newDialogAction = "testAction";
+    await store.dispatch("dialog/setDialogAction", newDialogAction);
+
+    // simulate clicking the cancel button of the dialog box
+    await simulateCancelClick(); // deepscan-disable-line
```

Comment:
I think you won't need to write this `// deepscan-disable-line` if you reference values and objects using `global`.
Like
`await global.simulateCancelClick()`

I had the same issue while implementing `Offline.js`
So importing the `Offline` npm package adds it to the global `window` object. All the methods are accessible by either using `window.checkOnlineStatus()` or directly like this `checkOnlineStatus()`.

Using the latter one gave me the same deepscan error, so I used `window.checkOnlineStatus()`. And this solution makes more sense as it's obvious to the reader that this is a method from the global scope so don't keep looking for this method in the current file

Reply:
do this for dummyItems, etc.